src/components/Analyze/charts/DoughnutChart.tsx

```diff
@@ -1,45 +1,52 @@
 import { AnalyzeContext } from '@/components/Analyze/AnalyzeContext'
-import { chartColors } from '@/Mock/Mock'
+import { ChartDataType } from '@/types/type'
 import { ResponsivePie } from '@nivo/pie'
 import { useContext, useEffect, useState } from 'react' // Import useEffect and useState
 
-interface ChartDataType {
-	id: string
-	label: string
-	value: number
-	color: string
-}
-
 export const DoughnutChart = () => {
-	const { doughnutChartData } = useContext(AnalyzeContext)
+	const { spendingData } = useContext(AnalyzeContext)
 	const [animatedData, setAnimatedData] = useState<ChartDataType[]>([])
 
+	const firstRow: ChartDataType[] = []
+	const secondRow: ChartDataType[] = []
+
+	spendingData?.map((each, idx) => {
+		if (idx <= spendingData.length / 2) firstRow.push(each.data)
+		else secondRow.push(each.data)
+	})
+
 	//차트가 변해야 애니메이션이 출력됨, default값을 넣고 0.05초뒤 필요한 값을 넣음
 	useEffect(() => {
-		if (!doughnutChartData) return
-		const initialData: ChartDataType[] = doughnutChartData.map((item) => ({
-			...item,
+		const initialData: ChartDataType[] = (spendingData || []).map((item) => ({
+			id: item.data.id,
+			label: item.data.label,
 			value: 0,
+			color: item.data.color,
 		}))
+
 		setAnimatedData(initialData)
 
 		const timeoutId = setTimeout(() => {
-			setAnimatedData(doughnutChartData)
+			setAnimatedData(
+				(spendingData || []).map((item) => ({
+					id: item.data.id,
+					label: item.data.label,
+					value: item.data.value,
+					color: item.data.color,
+				})),
+			)
```

Comment:
매핑 함수를 만들어 중복을 줄이는 건 어떠신가요? 또한 `spendingData`가 null이나 undefined & 데이터 존재 두가지 경우니 `??` 을 사용하는 것을 추천합니다.
```suggestion
const createChartData = (item: SpendingDataProps, value: number) => {
                        id: item.data.id,
			label: item.data.label,
			value,
			color: item.data.color,
 }
useEffect(() => {
     const initialData: ChartDataType[] = (spendingData ?? []).map((item) => createChartData(item, 0))

     setAnimatedData(initialData)

     const timeoutId = setTimeout(() => {
          setAnimatedData((spendingData ?? []).map((item) => createChartData(item, item.data.value)
          ))}, 50)

	return () => clearTimeout(timeoutId)
	}, [spendingData])
```

Reply:
좋은 영감 얻었습니다.